src/Bicep.Cli/Services/Evaluation.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Deployments.Core.Definitions.Schema;
+
+namespace Bicep.Cli.Services
+{
+    public class Evaluation
```

Comment:
Using a `record` rather than a `class` would be a good option here:
- Less boilerplate code required
- Everything is immutable by default

e.g. 
```csharp
namespace Bicep.Cli.Services;

public record Evaluation(
    Template? Template,
    InvalidOperationException? Error,
    ImmutableDictionary<string, bool> Assertions,
    ImmutableDictionary<string, bool> FailedAssertions)
{

    public bool Success => Error == null && (FailedAssertions.Count == 0);

    public bool Skip => Error != null;
}
```